src/ToggleButton/ToggleButton.js

```diff
@@ -0,0 +1,212 @@
+// @flow
+
+import React, { Component, Children, cloneElement, isValidElement } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import withStyles from '../styles/withStyles';
+import createStyleSheet from '../styles/createStyleSheet';
+import { fade } from '../styles/colorManipulator';
+import common from '../colors/common';
+
+/**
+ * Specification according to material.io
+ * - Have at least three toggle buttons in a group
+ * - Label buttons with text, an icon, or both
+ */
+
+export const styleSheet = createStyleSheet('MuiToggleButton', theme => ({
+  root: {
+    display: 'inline-block',
+    borderRadius: 2,
+    boxShadow: 'none',
+    overflow: 'hidden',
+    transition: theme.transitions.create(['background-color', 'box-shadow'], {
+      duration: theme.transitions.duration.short,
+    }),
+  },
+  active: {
+    backgroundColor: theme.palette.background.paper,
+    boxShadow: `0 3px 6px ${fade(common.black, 0.12)},
+        0 3px 6px ${fade(common.black, 0.12)}`,
+  },
+  toggleIcon: {
+    overflow: 'visible',
+  },
+}));
+
+class ToggleButton extends Component {
+  static defaultProps = {
+    exclusive: false,
+    toggleIcons: false,
+  };
+
+  state = {
+    active: false,
```

Comment:
I don't think that we need that state. As far as I can see it, it can be extracted from the `selectedOptions` property.